File: Examples/JAVA/SDK/src/main/java/com/aspose/slides/cloud/examples/convert/SplitPowerPointPresentations.java
package com.aspose.slides.cloud.examples.convert;

import java.nio.file.Path;

import com.aspose.slides.api.SlidesApi;
import com.aspose.slides.cloud.examples.common.Configuration;
import com.aspose.slides.cloud.examples.common.Utils;
import com.aspose.slides.model.SplitDocumentResponse;
import com.aspose.storage.api.StorageApi;

public class SplitPowerPointPresentations {

	public static void main(String[] args) {
		//ExStart: 1
		String fileName = "sample-input.pptx";
		int width = 0;
		int height = 0;
		int from = 2;
		int to = 3;
		String destFolder = "";
		String format = "png";
		String storage = "";
		String folder = "";

		Path inputFile = Utils.getPath(ConvertPowerPointDocuments.class, fileName);
		try
		{
			SlidesApi slidesApi = new SlidesApi(Configuration.apiKey, Configuration.appSID); 
			StorageApi storageApi = new StorageApi(Configuration.apiKey, Configuration.appSID, true);


		    // Upload source file to aspose cloud storage
		    storageApi.PutCreate(fileName, "", "", inputFile.toFile());

		    // Invoke Aspose.Slides Cloud SDK API to split powerpoint presentations
		    SplitDocumentResponse apiResponse = slidesApi.
		    										PostSlidesSplit(fileName, width, height, to, from, destFolder, format, storage, folder);

		    if (apiResponse != null)
		    {
		       
		        System.out.println("Split PowerPoint Presentations, Done!");
		        
		    }
		}
		catch (Exception ex)
		{
		    ex.printStackTrace();
		}
		//ExEnd: 1
	}

}

File: Examples/Python/SplitPowerPointPresentations.py
import asposeslidescloud
from asposeslidescloud.SlidesApi import SlidesApi
from asposeslidescloud.SlidesApi import ApiException

import asposestoragecloud
from asposestoragecloud.StorageApi import StorageApi
from asposestoragecloud.StorageApi import ResponseMessage

apiKey = "XXXXX" #sepcify App Key
appSid = "XXXXX" #sepcify App SID
apiServer = "http://api.aspose.com/v1.1"
data_folder = "../../data/"

#Instantiate Aspose Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, True)
storageApi = StorageApi(storage_apiClient)
#Instantiate Aspose Slides API SDK
api_client = asposeslidescloud.ApiClient.ApiClient(apiKey, appSid, True)
slidesApi = SlidesApi(api_client);

#set input file name
name =  "sample-input.pptx"
ffrom = 2
to = 3
format = "png"


try:
    #upload file to aspose cloud storage
    response = storageApi.PutCreate(name, data_folder + name)

    #invoke Aspose.Slides Cloud SDK API to split all or specific slides of a presentation file to specify format
    response = slidesApi.PostSlidesSplit(name, ffrom=ffrom, to=to, format=format)

    if response.Status == "OK":
        for slideUri in response.SplitResult.Slides:
            urlString = slideUri.Href
            slidName = urlString.split('/')[-1]
            #download presentation from cloud storage
            response = storageApi.GetDownload(Path=slidName)
            outfilename = "c:/temp/" + slidName
            with open(outfilename, 'wb') as f:
                        for chunk in response.InputStream:
                            f.write(chunk)
except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message
